lib/api/src/externals/function.rs

```diff
@@ -73,24 +74,22 @@ fn build_export_function_metadata<Env>(
         &'a mut Env,
         &'a crate::Instance,
     ) -> Result<(), crate::HostEnvInitError>,
-) -> (*mut std::ffi::c_void, ExportFunctionMetadata)
+) -> (*mut c_void, ExportFunctionMetadata)
 where
     Env: Clone + Sized + 'static + Send + Sync,
 {
     let import_init_function_ptr = Some(unsafe {
-        std::mem::transmute::<fn(_, _) -> Result<(), _>, fn(_, _) -> Result<(), _>>(
-            import_init_function_ptr,
-        )
+        std::mem::transmute::<_, ImportInitializerFuncPtr>(import_init_function_ptr)
```

Comment:
Transmuting here is simplified and clearer, _and_ most importantly, it uses an existing type alias, which prevents future bugs, and improves code navigation.